chromium_src/net/base/url_util.cc

```diff
@@ -80,4 +82,16 @@ bool ParseAuthHostAndPort(base::StringPiece input,
   return true;  // Success.
 }
 
+std::string URLToEphemeralStorageDomain(const GURL& url) {
+  std::string domain = registry_controlled_domains::GetDomainAndRegistry(
+      url, registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES);
+
+  // GetDomainAndRegistry might return an empty string if this host is an IP
+  // address or a file URL.
+  if (domain.empty())
+    domain = url::Origin::Create(url.GetOrigin()).Serialize();
```

Comment:
I don't believe this is the correct behavior. You should create a URL from the domain even if it's empty and in turn Origin will create an opaque origin for those entries

Reply:
This would mean that all sites that are accessed with IP addresses share the same ephemeral storage. If that the behaviour that we want, I think it makes sense to make that change in a separate PR that also adds tests for that functionality. Since this PR is just about adding support for ephemeral `document.cookie`, I'm not sure it belongs here.

Reply:
no, opaque origins are unique by design, but I think actually it would cause the opposite problem where the same IP in more than one tab would go into different ephemeral storage domains. Since url.Getorigin will return an empty url (and hence create an opaque origin) for file urls, I believe you need to call `SerializeWithNonce`

Reply:
what is `domain` for a file url with this code?